_includes/tutorials/filtering/flinksql/code/src/test/java/io/confluent/developer/FlinkSqlFilteringTest.java

```diff
@@ -0,0 +1,37 @@
+package io.confluent.developer;
+
+
+import org.apache.flink.table.api.TableResult;
+import org.junit.Test;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+
+public class FlinkSqlFilteringTest extends AbstractFlinkKafkaTest {
+
+  @Test
+  public void simpleSelect() throws Exception {
+    // create base movie sales table and aggregation table, and populate with test data
+    streamTableEnv.executeSql(getResourceFileContents("create-all-publications.sql.template",
+       Optional.of(kafkaPort),Optional.of(schemaRegistryPort))).await();
+    streamTableEnv.executeSql(getResourceFileContents("create-publications-by-author.sql.template",
+       Optional.of(kafkaPort),Optional.of(schemaRegistryPort))).await();
+  
+
+    // We can't call await() on this result since it won't return. In Flink 17 and later this can change
+    // by setting 'scan.bounded.mode' = 'latest-offset' in the movie_sales CREATE TABLE statement, which will
+    // cause this INSERT to terminate once the latest offset of movie_sales table is reached.
+    streamTableEnv.executeSql(getResourceFileContents("populate-publication-events.sql"));
+
```

Comment:
also need to execute `populate-publication-events-by-author.sql`

Reply:
good catch -fixed